manta-accounting/src/wallet/signer/functions.rs

```diff
@@ -1039,6 +1132,33 @@ where
     Ok(result)
 }
 
+/// Signs a transaction which consolidates the assets in `request`,
+/// generating transfer posts without releasing resources.
+#[inline]
+pub fn consolidate<C>(
+    parameters: &SignerParameters<C>,
+    accounts: Option<&AccountTable<C>>,
+    assets: &C::AssetMap,
+    utxo_accumulator: &mut C::UtxoAccumulator,
+    request: ConsolidationPrerequest<C>,
+    rng: &mut C::Rng,
+) -> Result<SignResponse<C>, SignError<C>>
+where
+    C: Configuration,
+    C::Identifier: PartialEq,
+{
+    let result = consolidate_internal(
+        parameters,
+        accounts.ok_or(SignError::MissingSpendingKey)?,
+        assets,
+        utxo_accumulator,
+        request.try_into()?,
+        rng,
+    )?;
+    utxo_accumulator.rollback();
```

Comment:
To generate the transaction, you need to create receivers that you may need to spend again, so you have to update the internal UTXO accumulator. But that doesn't correspond to the actual ledger state (e.g. someone else may have minted some UTXOs while you were generating the proofs or your transaction may be rejected) so you rollback to the original state after signing. Only after syncing you want to update the accumulator in a definite manner (calling `commit` instead) because that's actual ledger data that you're importing.